src/devices/device_models/common/rateofchange_constraints.jl

```diff
@@ -0,0 +1,70 @@
+function device_linear_rateofchange(ps_m::CanonicalModel, 
+                                    rate_data::Array{Tuple{String,NamedTuple{(:up, :down),Tuple{Float64,Float64}}},1}, 
```

Comment:
I would prefer to use a alias for the tuple portion of the type signature and use `Vector{T}` instead of `Array{T, 1}`
```
const UpDownRate = Tuple{String,NamedTuple{(:up, :down),Tuple{Float64,Float64}}}
...
rate_data::Vector{UpDownRate}
...
```